shared/src/main/scala/mlscript/Typer.scala

```diff
@@ -1024,6 +1081,18 @@ class Typer(var dbg: Boolean, var verbose: Bool, var explainErrors: Bool, val ne
       case pat if ctx.inPattern =>
         err(msg"Unsupported pattern shape${
           if (dbg) " ("+pat.getClass.toString+")" else ""}:", pat.toLoc)(raise)
+      case Lam(pat, body) if ctx.inQuote =>
+        println(s"TYPING QUOTED LAM")
+        ctx.qnest(true).poly { newCtx =>
```

Comment:
I still don't understand the logic of "we need to clean `qenv` and `fvars`" here. What if this lambda is _deep_ inside a quote, like is `y => ...` in program `code"let x = ... in y => ..."`? Isn't `ctx.inQuote` going to be true here?

Reply:
> I still don't understand the logic of "we need to clean `qenv` and `fvars`" here. What if this lambda is _deep_ inside a quote, like is `y => ...` in program `code"let x = ... in y => ..."`? Isn't `ctx.inQuote` going to be true here?
For `y => ...`, `ctx.inQuote` is true.  I further added some comments for `qenv` and `fvars`.